### PR TITLE
Response headers regex

### PR DESCRIPTION
**Pull request by dimpiax**
Optimized regex for retrieving headers pair.
Saved bytes: **7**.

But we can save **1** byte more, if we will consider on next situation. 
Do we have real case when `getAllResponseHeaders` returns not strict format of pair?  
Example: `X-Foo: bar\nX-Foo:baz`

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,39 +1,18 @@
 export default typeof fetch=='function' ? fetch : function(url, options) {
 	options = options || {};
-	return new Promise( (resolve, reject) => {
-		let request = new XMLHttpRequest();
+	return new Promise((resolve, reject) => {
+		const response = () => {
```
Comment (developit) [inline src/index.js]: probably best to leave this as a function declaration so the name doesn't get duplicated.

**Comment by developit** [comment]
I sort-of wish the regex and refactor changes were in separate PRs 🙊
Regex one is a quick merge and seems good to me (regarding your point about normalized values from `getAllResponseHeaders()` - I think we'll need to double check in IE10/11 and Safari since that's really the main two places this polyfill actually gets run).  Refactoring stuff doesn't seem to really affect code size and is more of a stylistic preference.

**Comment by dimpiax** [comment]
@developit disagree, check the new commit with current syntax – I saved 1 byte, 😂.
Currently with proposed regex, gzipped size is 500bytes.

**Comment by dimpiax** [comment]
@developit any issues about PR?

**Comment by developit** [comment]
Ahh no, sorry, I just let this slip. I had meant to do some browser testing.

**Comment by dimpiax** [comment]
Any status?

**Pull request closed without merging**